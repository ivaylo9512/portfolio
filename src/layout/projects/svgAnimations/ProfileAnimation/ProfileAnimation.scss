@use '@sassUtils/classGenerators/colors';

.profile-animation{
    fill: var(--clr-profile-secondary);

    .shadow {
        filter: url(#profile-shadow);
    }

    .lines rect,
    .squares rect,
    .circles circle{
        transform-box: fill-box;
    }

    .circles circle{
        transform-origin: center;
    }

    &:hover{
        .lines rect{
            width: 0;
        }

        .squares rect{
            transform: scale(0);
        }

        .circles circle{
            transform: scale(0);
        }

        .lines rect:nth-child(1){
            animation: ProfileWidthTransition 3s 0.5s infinite;
        }

        .lines rect:nth-child(2){
            animation: ProfileWidthTransitionFading 3s 1s infinite;
        }

        .lines rect:nth-child(3){
            animation: ProfileWidthTransitionSecond 3s 1.5s infinite;
        }

        .squares rect:nth-child(1){
            animation: bounceIn 3s 0.5s infinite cubic-bezier(0.215, 0.610, 0.355, 1.000);
        }

        .squares rect:nth-child(2){
            animation: bounceInDelayed 3s 1s infinite cubic-bezier(0.215, 0.610, 0.355, 1.000);
        }

        .circles circle{
            animation: scaleIn 3s 0.5s infinite cubic-bezier(0.215, 0.610, 0.355, 1.000);
        }
    }

    @keyframes ProfileWidthTransition {
        0%{
            width: 0;
        }
        50%, 83.38%{
            width: 36px;
        }
        83.39%, 100%{
            width: 0;
        }
    }

    @keyframes ProfileWidthTransitionFading {
        0%{
            width: 0;
        }
        50%, 66.66%{
            width: 36px;
        }
        66.67%, 100%{
            width: 0;
        }
    }

    @keyframes ProfileWidthTransitionSecond {
        0%{
            width: 0;
        }
        50%{
            width: 36px;
        }
        50.01%, 100%{
            width: 0;
        }
    }

    @keyframes bounceIn {
        0%{
            transform: scale(0.65) rotate(15deg);
        }
        50%, 80%{
            transform: scale(1) rotate(0deg);
        }
        100%{
            transform: scale(0.65) rotate(15deg);
        }
    }

    @keyframes bounceInDelayed {
        0%{
            transform: scale(0.65) rotate(15deg);

        }
        50%, 66.66%{
            transform: scale(1) rotate(0deg);
        }
        100%{
            transform: scale(0.65) rotate(15deg);

        }
    }

    @keyframes scaleIn {
        0%{
            transform: scale(0.6);
        }
        50%, 80%{
            transform: scale(1);
        }
        100%{
            transform: scale(0.6);
        }
    }
}