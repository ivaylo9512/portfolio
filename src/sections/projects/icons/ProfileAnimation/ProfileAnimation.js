const ProfileAnimation = () => {
    return(
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 274.42 138.03">
            <g id="Layer_2" data-name="Layer 2">
                <g id="Layer_53" data-name="Layer 53">
                    <circle class="cls-1" cx="74.85" cy="65.06" r="6.68" />
                </g>
                <g id="Layer_38" data-name="Layer 38">
                    <rect class="cls-2" x="49.46" y="35.17" width="134.6" height="102.36" rx="6.63" />
                    <circle class="cls-2" cx="239.09" cy="53.21" r="34.83" />
                    <circle class="cls-3" cx="74.85" cy="65.06" r="17.86" />
                    <circle class="cls-2" cx="74.85" cy="65.06" r="11.79" />
                    <rect class="cls-3" x="57.26" y="92.88" width="36" height="3.24" rx="1" />
                    <rect class="cls-3" x="57.26" y="103.69" width="36" height="3.24" rx="1" />
                    <rect class="cls-3" x="57.26" y="114.26" width="36" height="3.24" rx="1" />
                    <rect class="cls-3" x="143.1" y="45.38" width="13" height="13" rx="3" />
                    <rect class="cls-3" x="160.76" y="45.79" width="13" height="13" rx="3" />
                    <circle class="cls-3" cx="128.47" cy="112.25" r="6.47" />
                    <circle class="cls-2" cx="128.47" cy="112.25" r="4.27" />
                    <circle class="cls-3" cx="17.86" cy="17.86" r="17.86" />
                    <circle class="cls-2" cx="17.86" cy="17.86" r="11.79" />
                    <circle class="cls-3" cx="147.1" cy="112.16" r="6.47" />
                    <circle class="cls-2" cx="147.1" cy="112.16" r="4.27" />
                    <circle class="cls-3" cx="165.1" cy="112.16" r="6.47" />
                    <circle class="cls-2" cx="165.1" cy="112.16" r="4.27" />
                </g>
            </g>
        </svg>
    )
}

export default ProfileAnimation