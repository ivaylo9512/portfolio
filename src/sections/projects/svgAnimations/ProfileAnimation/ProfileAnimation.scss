@use '@sassUtils/projectsClasses/colors';

.profile-animation{
    fill: var(--clr-profile-secondary);

    .shadow {
        filter: url(#profile-shadow);
    }
}