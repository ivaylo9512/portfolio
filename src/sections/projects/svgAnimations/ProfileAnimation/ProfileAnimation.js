import ProjectsShadow from 'sections/projects/svgFilters/ProjectsShadow';
import './ProfileAnimation.scss';

const ProfileAnimation = ({ className }) => {
    return(
        <svg className={className} xmlns='http://www.w3.org/2000/svg' viewBox='0 0 274.42 138.03'>
            <defs>
                <ProjectsShadow id='profile-shadow'/>
            </defs>
            <circle class='fill-primary shadow' cx='37.86' cy='137.86' r='35.86'></circle>
            <circle class='fill-primary shadow' cx='190.09' cy='31.21' r='46.83' ></circle>
            <rect class='fill-primary shadow' x='49.46' y='35.17' width='134.6' height='102.36' rx='6.63'></rect>
            <circle cx='74.85' cy='65.06' r='17.86'></circle>
            <circle cx='147.1' cy='112.16' r='4.27'></circle>
            <rect x='57.26' y='92.88' width='36' height='3.24' rx='1'></rect>
            <rect x='57.26' y='103.69' width='36' height='3.24' rx='1'></rect>
            <rect x='57.26' y='114.26' width='36' height='3.24' rx='1'></rect>
            <rect x='143.1' y='45.38' width='13' height='13' rx='3'></rect>
            <rect x='160.76' y='45.79' width='13' height='13' rx='3'></rect>
            <circle cx='128.47' cy='112.25' r='4.27'></circle>
            <circle cx='165.1' cy='112.16' r='4.27'></circle>
        </svg>
    )
}

export default ProfileAnimation