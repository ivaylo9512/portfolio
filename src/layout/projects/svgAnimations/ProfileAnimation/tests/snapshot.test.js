import { shallow } from 'enzyme';
import ProfileAnimation from '../ProfileAnimation'

describe('ProfileAnimation snapshot tests', () => {
    const createWrapper = () => shallow(
        <ProfileAnimation />
    )

    it('should match snapshot', () => {
        const wrapper = createWrapper();

        expect(wrapper).toMatchSnapshot();
    })
})