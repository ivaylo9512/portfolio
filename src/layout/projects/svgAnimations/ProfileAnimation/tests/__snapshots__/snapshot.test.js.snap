// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAnimation snapshot tests should match snapshot 1`] = `
<svg
  viewBox="0 0 274.42 138.03"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <ProjectsShadow
      id="profile-shadow"
    />
  </defs>
  <circle
    className="fill-primary shadow"
    cx="37.86"
    cy="137.86"
    r="35.86"
  />
  <circle
    className="fill-primary shadow"
    cx="190.09"
    cy="31.21"
    r="46.83"
  />
  <rect
    className="fill-primary shadow"
    height="102.36"
    rx="6.63"
    width="134.6"
    x="49.46"
    y="35.17"
  />
  <circle
    cx="74.85"
    cy="65.06"
    r="17.86"
  />
  <circle
    cx="147.1"
    cy="112.16"
    r="4.27"
  />
  <rect
    height="3.24"
    rx="2"
    width="36"
    x="57.26"
    y="92.88"
  />
  <rect
    height="3.24"
    rx="2"
    width="36"
    x="57.26"
    y="103.69"
  />
  <rect
    height="3.24"
    rx="2"
    width="36"
    x="57.26"
    y="114.26"
  />
  <rect
    height="13"
    rx="3"
    width="13"
    x="143.1"
    y="45.38"
  />
  <rect
    height="13"
    rx="3"
    width="13"
    x="160.76"
    y="45.79"
  />
  <circle
    cx="128.47"
    cy="112.25"
    r="4.27"
  />
  <circle
    cx="165.1"
    cy="112.16"
    r="4.27"
  />
</svg>
`;
